### PR TITLE
[#70] As a user, I can see the list of favorited articles from the profile screen

### LINKED ISSUE #70: As a user, I can see the list of favorited articles from the profile screen
## Why
The users will be able to see the list of favorited articles of an author in the `Profile` screen regardless whether they are logged in or not with data provided from server.

## Acceptance Criteria
- [ ] Implement the API for getting favorited articles data via this [endpoint](https://github.com/gothinkster/realworld/tree/master/api#list-articles).
- [ ] Provide `favorited` query parameter to filter the results accordingly the author's username.
- [ ] Setup the page size to `25`.
- [ ] Only load the first latest page and no need pagination support.
- [ ] Create a usecase for getting author favorited articles for later usage in integrate task.
- [ ] Parse the response into the corresponding models, here is an example response for the list of author favorited articles:
```
{
  "articles":[{
    "slug": "how-to-train-your-dragon",
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "It takes a Jacobian",
    "tagList": ["dragons", "training"],
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "favorited": false,
    "favoritesCount": 0,
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    }
  }, {
    "slug": "how-to-train-your-dragon-2",
    "title": "How to train your dragon 2",
    "description": "So toothless",
    "body": "It a dragon",
    "tagList": ["dragons", "training"],
    "createdAt": "2016-02-18T03:22:56.637Z",
    "updatedAt": "2016-02-18T03:48:35.824Z",
    "favorited": false,
    "favoritesCount": 0,
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    }
  }],
  "articlesCount": 2
}
```

### PR DESCRIPTION
**Pull request by markgravity**
Resolved #70

## What happened

Add GetFavourited Articles UseCase

## Insight

n/a

## Proof Of Work

Request
```
*** Overview *** 
URL https://conduit.productionready.io/api/articles?favorited=markg&limit=25
Method GET
Response Code -
Request Start Time Sep 28 2021 - 09:58:52
Duration 219ms

*** Request Header *** 
Accept-Language en;q=1.0
Accept-Encoding br;q=1.0, gzip;q=0.9, deflate;q=0.8
User-Agent Nimble Medium - Staging/1.2.0 (co.nimblehq.nimble-medium-stag; build:1; iOS 14.5.0) Alamofire/5.4.4


*** Request Body *** 
-

*** Response Header *** 
-

*** Response Body *** 
-

------------------------------------------------------------------------
------------------------------------------------------------------------
------------------------------------------------------------------------
```

Subcribe Log
```
2021-09-28 09:58:52.867: GetFavouritedArticlesUseCase -> subscribed
2021-09-28 09:58:54.261: GetFavouritedArticlesUseCase -> Event next([NimbleMedium.DecodableArticle(slug: "Welcome-to-RealWorld-project", title: "Welcome to RealWorld project", description: "Exemplary fullstack Medium.com clone powered by React, Angular, Node, Django, and many more", body: "See how the exact same Medium.com clone (called Conduit) is built using different frontends and backends. Yes, you can mix and match them, because they all adhere to the same API spec", tagList: ["introduction", "welcome"], createdAt: 2021-09-25 18:17:54 +0000, updatedAt: 2021-09-25 18:17:54 +0000, favorited: false, favoritesCount: 21, apiAuthor: NimbleMedium.DecodableProfile(username: "Gerome", bio: nil, image: Optional("https://realworld-temp-api.herokuapp.com/images/demo-avatar.png"), _following: DefaultCodable.Default<DefaultCodable.False>(wrappedValue: false)))])
2021-09-28 09:58:54.261: GetFavouritedArticlesUseCase -> Event completed
2021-09-28 09:58:54.261: GetFavouritedArticlesUseCase -> isDisposed
```

### CODE REVIEW

File: NimbleMediumTests/Sources/Specs/Domain/UseCases/ListArticlesUseCaseSpec.swift [hunk 1]
```diff
@@ -21,7 +21,7 @@ final class ListArticlesUseCaseSpec: QuickSpec {
         var scheduler: TestScheduler!
         var disposeBag: DisposeBag!
 
-        describe("an ArticleRepository") {
+        describe("an ListArticlesUseCase") {
```
Comment (minhnimble) [inline NimbleMediumTests/Sources/Specs/Domain/UseCases/ListArticlesUseCaseSpec.swift:24]: please ignore this change, I have already added some code refactors for this already 🙏

File: NimbleMedium/Sources/Injection/Resolver+Injection.swift [hunk 2]
```diff
@@ -61,6 +61,9 @@ extension Resolver: ResolverRegistering {
         register {
             GetArticleCommentsUseCase(articleCommentRepository: resolve())
         }.implements(GetArticleCommentsUseCaseProtocol.self)
+        register {
+            GetFavouritedArticlesUseCase(articleRepository: resolve())
+        }.implements(GetFavouritedArticlesUseCaseProtocol.self)
```
Comment (minhnimble) [inline NimbleMedium/Sources/Injection/Resolver+Injection.swift:66]: Please help sort the new use case in this function, making sure it follows alphabetically order for easy managing later on 🙏

**Review by minhnimble — COMMENTED** [review]
If possible please rebase your branch on my PR here: https://github.com/nimblehq/nimble-medium-ios/pull/156 for grouping use cases into correct folders. I created the folders for better organizing as following:

Screen Shot 2021-09-28 at 10 17 19

**Review by suho — APPROVED** [review]

File: NimbleMediumTests/Sources/Specs/Domain/UseCases/Article/ListArticlesUseCaseSpec.swift [hunk 3]
```diff
@@ -21,7 +21,7 @@ final class ListArticlesUseCaseSpec: QuickSpec {
         var scheduler: TestScheduler!
         var disposeBag: DisposeBag!
 
-        describe("an ListArticlesUseCase") {
+        describe("an ArticleRepository") {
```
Comment (minhnimble) [inline NimbleMediumTests/Sources/Specs/Domain/UseCases/Article/ListArticlesUseCaseSpec.swift:24]: we don't need this change 😅

Comment (markgravity) [reply NimbleMediumTests/Sources/Specs/Domain/UseCases/Article/ListArticlesUseCaseSpec.swift]: @minhnimble my bad, I forgot it after rebase 😅
Updated 🙏

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

**Comment by minhnimble** [comment]
Please merge this PR after #156 gets merged 🙏

**Comment by minhnimble** [comment]
Just curious, we don't have any sample request/response log? how can you verify if the API works then 😅
Btw we can remove the word `[Backend]` in the PR header because there is the label tagged with the PR already 🙏

**Comment by markgravity** [comment]
> Just curious, we don't have any sample request/response log? how can you verify if the API works then 😅
> Btw we can remove the word `[Backend]` in the PR header because there is the label tagged with the PR already 🙏

It has the same response with list article api which already added a json file 😁

**Pull request merged**